runtime/queries/metricsview_comparison_toplist.go

```diff
@@ -228,20 +239,29 @@ func timeRangeClause(timeRange *runtimev1.TimeRange, td string, args *[]any) str
 	return clause
 }
 
-func (q *MetricsViewComparisonToplist) buildMetricsTopListSQL(mv *runtimev1.MetricsView, dialect drivers.Dialect, policy *runtime.ResolvedMetricsViewSecurity) (string, []any, error) {
+func (q *MetricsViewComparisonToplist) buildMetricsTopListSQL(mv *runtimev1.MetricsView, dialect drivers.Dialect, policy *runtime.ResolvedMetricsViewSecurity, forExport bool) (string, []any, error) {
 	ms, err := resolveMeasures(mv, q.InlineMeasures, q.MeasureNames)
 	if err != nil {
 		return "", nil, err
 	}
 
+	q.measureNameToLabel = make(map[string]string)
+	for _, m := range ms {
+		if m.Label != "" && forExport {
+			q.measureNameToLabel[m.Name] = m.Label
+		} else {
+			q.measureNameToLabel[m.Name] = m.Name
+		}
+	}
+
 	colName, err := metricsViewDimensionToSafeColumn(mv, q.DimensionName)
 	if err != nil {
 		return "", nil, err
 	}
 
 	selectCols := []string{colName}
 	for _, m := range ms {
-		expr := fmt.Sprintf(`%s as %s`, m.Expression, safeName(m.Name))
+		expr := fmt.Sprintf(`%s as %s`, m.Expression, safeName(q.measureNameToLabel[m.Name]))
```

Comment:
Measure labels are not guaranteed to be unique. They can't be used inside the query for ordering/filtering/etc.

They should only be used at the export stage or in the query passed to `COPY TO` (where DuckDB will add numbers to disambiguate).